src/requests/utils.ts

```diff
@@ -0,0 +1,45 @@
+import * as querystring from 'querystring'
+
+export const getNormalizedEncodedURI = (uri: string) => {
+  const [pathname, search] = uri.split('?')
+  const escapedSearch = search ? querystring.escape(search) : ''
+
+  return encodeURI(escapedSearch ? `${pathname}?${escapedSearch}` : pathname)
+}
+
+export const sortHeaderKeys = (keyA: string, keyB: string) => (keyA > keyB ? 1 : -1)
+
+const normalizeHeaderKey = (key: string) => key.toLowerCase().trim()
+const normalizeHeaderValue = (value: string) => value.trim()
```

Comment:
Removed the URI encoding because unnecessary [according to the specification](https://tools.ietf.org/html/rfc7230#section-3.2.4). 

Plus, it's unstable (as in, you cannot compose encodings) which defeats the purpose of normalization.